File: CS/TokenEdit_Glyph_InEndOfToken/CustomTokenEdit/CustomTokenEditValidationController.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Repository;
namespace TokenEdit_Glyph_InEndOfToken.CustomTokenEdit
{
    public class CustomTokenEditValidationController : TokenEditValidationController
    {
        public CustomTokenEditValidationController(RepositoryItemTokenEdit properties) : base(properties) { }

        protected override TokenEditToken CreateToken(string description, object value)
        {
            CustomTokenEditToken token = new CustomTokenEditToken(description, value, Properties as RepositoryItemCustomTokenEdit);
            return token;
        }
    }
}

File: VB/CustomTokenEdit/CustomTokenEditValidationController.vb
Imports DevExpress.XtraEditors
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors.Repository
Namespace Example.CustomTokenEdit
	Public Class CustomTokenEditValidationController
		Inherits TokenEditValidationController

		Public Sub New(ByVal properties As RepositoryItemTokenEdit)
			MyBase.New(properties)
		End Sub

		Protected Overrides Function CreateToken(ByVal description As String, ByVal value As Object) As TokenEditToken
			Dim token As New CustomTokenEditToken(description, value, TryCast(Properties, RepositoryItemCustomTokenEdit))
			Return token
		End Function
	End Class
End Namespace
